src/main/java/com/opentable/kafka/partition/ManualKafkaPartitioningStrategy.java

```diff
@@ -0,0 +1,132 @@
+            if (((instanceCount > partitionCount) && (brokerConfig.isEnabled()))) {
+                throw new IllegalStateException("More instances than partitions - probably a misconfiguration?");
+            }
+            // Calculate distribution of instances to partitions
+            Multimap<Integer, Integer> map = getInstanceToPartitionMap(partitionCount, instanceCount);
+            this.partitionList = ImmutableList.copyOf(map.get(instanceNumber - 1));
+            LOG.info("Partition count {}, instance number {}, instance count {} partitions to assign {}", partitionCount, instanceNumber, instanceCount, partitionList);
+        }
+
+        @VisibleForTesting
+        Multimap<Integer, Integer> getInstanceToPartitionMap(final int partitionCount, final int instanceCount) {
+            final Multimap<Integer, Integer> map = HashMultimap.create();
+            for (int partition = 0; partition < partitionCount; partition++) {
+                int assignedInstance = partition % instanceCount;
+                map.put(assignedInstance, partition);
+            }
+            return map;
+        }
+
+        // On local or non mesos always return 1
+        private int calculateTotalInstances() {
+            return appinfo != null && appinfo.getInstanceCount() != null ? appinfo.getInstanceCount(): 1;
+        }
+
+        private int calculateInstanceNumber() {
+            return appinfo != null && appinfo.getInstanceNumber() != null ? appinfo.getInstanceNumber() : 1;
+        }
+
+    /*
+        Use Kafka admin api to determine the number of partitions for this topic
+    */
+    private int numberOfPartitions(final String topic, final String brokerList) throws ExecutionException, InterruptedException {
+            if (!brokerConfig.isEnabled()) {
+                return 1;
+            }
+            final AdminClient client = makeAdminClient(brokerList);
+            final DescribeTopicsResult result = client.describeTopics(Lists.newArrayList(topic));
+            final TopicDescription topicDescription = result.all().get().get(topic);
+            return topicDescription.partitions().size();
+    }
+
+    private AdminClient makeAdminClient(final String brokerList) {
+        final Properties props = new Properties();
+        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+        return AdminClient.create(props);
+    }
+
+    /**
+     * Return the Partitions assigned to this instance
+     */
+    @Override
+    public List<Integer> getPartitionList() {
+        return partitionList;
+    }
+
+    /**
+        Use this to create a bean providing the basic configuration needed
+     */
+    public static class BrokerConfig {
+        private final String brokerList; // comma delimited list of brokers
+        private final String topic; // topic
+        private final boolean enabled; // is it enabled? Should be true for "normal" ops, false for when testing without kafka
```

Comment:
This seems like too much policy, too little mechanism.  How about we omit this enabled idea from the common library?

Reply:
Problem is you need it or the constructor fails.

Reply:
I've streamlined it but it's still there for one use - in getPartitionCount(). Basically the question is "How can I know whether to call this and fail". Clearly in normal deployed circumstances or integration tests I should fail. But in unit tests etc, I should not.

I did add the strategy and we could in theory add that there, but...